### PR TITLE
Updates of mRICH positioning to avoid overlaps

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This PR updates the module parameters for the mRICH and packs as many as possible in the envelope volume.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

### CODE REVIEW

**Comment by wdconinc** [comment]
As many smaller modules as reasonable...

Before (`main` branch, with larger modules):
image

Before (`mrich` branch, with smaller modules):
image

After (`mrich` branch, with closest packing):
image

File: compact/mrich.xml
Comment (wdconinc) [inline compact/mrich.xml:8]: @c-dilks This is how I'm pushing the MRICH up against zmax. Wasn't as cumbersome as I had feared (and definitely avoided having to mess with jinja).

File: compact/mrich.xml
Comment (wdconinc) [inline compact/mrich.xml:28]: Remove commented line.
```suggestion
```

File: compact/mrich.xml
Comment (c-dilks) [reply compact/mrich.xml:8]: Makes sense, I will update #158 to be consistent. Btw, I get `172.470` for `BackwardPIDRegion_zmax` here, but the menagerie says `172.7`. It's not a big enough difference to matter, at least for the pfRICH, so I'll leave it up to you if you want to update it here.

Comment (c-dilks) [reply compact/mrich.xml:8]: Just need to change `CentralTrackingRegionN_zmax`

**Review by osbornjd — COMMENTED** [review]
Looks good to me from an integration standpoint, everything seems to be within the menagerie as far as I can tell. I defer to @c-dilks for full approval as the RICH expert.

File: compact/mrich.xml
Comment (wdconinc) [reply compact/mrich.xml:8]: I'm deferring to the `BackwardPIDRegion_zmax` as provided by the parametrization, so I'm not going to hard code it here.

**Comment by c-dilks** [comment]
Looks good to me too. Once this PR is merged I'm going to change `BackwardPIDRegion_rmax` from 64->63 cm in #158 and rebase, so both the pfRICH and mRICH are consistent with the menagerie. The only side effect is on the endcap TOF radius, so that's fine.

**Review by c-dilks — DISMISSED** [review]

**Comment by c-dilks** [comment]
> I'm going to change BackwardPIDRegion_rmax from 64->63 cm
Hmm... after looking at the mRICH in `jsroot` I probably shouldn't....

**Comment by c-dilks** [comment]
In that case, why don't we just change `BackwardPIDRegion_rmax` from 64 to 63 in `definitions.xml`? It's _only_ used by `mrich.xml`, `pfrich.xml`, and `tof_endcap.xml`, and no other parameters in `definitions.xml` depend on it.

**Comment by wdconinc** [comment]
Yes, that's the goal but I wanted to keep this focused on the mRICH first.

**Comment by wdconinc** [comment]
image

**Comment by wdconinc** [comment]
Cannot auto-update because of conflicts.

**Review by c-dilks — APPROVED** [review]

**Pull request merged**
